cosmology/GravityManager.h

```diff
@@ -0,0 +1,367 @@
+    using LoadBalancer_t= LoadBalancer<T, Dim>;
+    using Base= ippl::ParticleBase<ippl::ParticleSpatialLayout<T, Dim>>;
+
+public:
+    /**
+     * @brief Constructor for GravityManager.
+     * 
+     * @param totalP_ Total number of particles.
+     * @param nt_ Number of time steps.
+     * @param nr_ Number of grid points in each dimension.
+     * @param lbt_ Load balance threshold.
+     * @param solver_ Solver type.
+     * @param stepMethod_ Step method type.
+     */
+    GravityManager(size_type totalP_, int nt_, Vector_t<int, Dim>& nr_, double lbt_, std::string& solver_, std::string& stepMethod_)
+        : ippl::PicManager<T, Dim, ParticleContainer<T, Dim>, FieldContainer<T, Dim>, LoadBalancer<T, Dim>>()
+        , totalP_m(totalP_)
+        , nt_m(nt_)
+        , nr_m(nr_)
+        , lbt_m(lbt_)
+        , solver_m(solver_)
+        , stepMethod_m(stepMethod_){}
+    
+    /**
+     * @brief Destructor for GravityManager.
+     */
+    ~GravityManager(){}
+
+
+public:
+    /**
+     * @brief Folder for initial conditions.
+     */
+    std::string folder;
+    
+
+    /**
+     * @brief Get the total number of particles.
+     * 
+     * @return Total number of particles.
+     */
+    size_type getTotalP() const { return totalP_m; }
+
+    /**
+     * @brief Set the total number of particles.
+     * 
+     * @param totalP_ Total number of particles.
+     */
+    void setTotalP(size_type totalP_) { totalP_m = totalP_; }
+
+    /**
+     * @brief Get the number of time steps.
+     * 
+     * @return Number of time steps.
+     */
+    int getNt() const { return nt_m; }
+
+    /**
+     * @brief Set the number of time steps.
+     * 
+     * @param nt_ Number of time steps.
+     */
+    void setNt(int nt_) { nt_m = nt_; }
+
+    /**
+     * @brief Get the solver type.
+     * 
+     * @return Solver type.
+     */
+    const std::string& getSolver() const { return solver_m; }
+
+    /**
+     * @brief Set the solver type.
+     * 
+     * @param solver_ Solver type.
+     */
+    void setSolver(const std::string& solver_) { solver_m = solver_; }
+
+    /**
+     * @brief Get the load balance threshold.
+     * 
+     * @return Load balance threshold.
+     */
+    double getLoadBalanceThreshold() const { return lbt_m; }
+
+    /**
+     * @brief Set the load balance threshold.
+     * 
+     * @param lbt_ Load balance threshold.
+     */
+    void setLoadBalanceThreshold(double lbt_) { lbt_m = lbt_; }
+
+    /**
+     * @brief Get the step method type.
+     * 
+     * @return Step method type.
+     */
+    const std::string& getStepMethod() const { return stepMethod_m; }
+
+    /**
+     * @brief Set the step method type.
+     * 
+     * @param stepMethod_ Step method type.
+     */
+    void setStepMethod(const std::string& stepMethod_) { stepMethod_m = stepMethod_; }
+
+    /**
+     * @brief Get the number of grid points in each dimension.
+     * 
+     * @return Number of grid points in each dimension.
```

Comment:
I would say "Vector type of size Dim with the number of grid points in each dimension"